test/api/oran_o2ims_suite_test.go

```diff
@@ -0,0 +1,324 @@
+package main_test
+
+import (
+	"bytes"
+	"os"
+	"testing"
+	"crypto/tls"
+	"io"
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestOranO2ims(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "OranO2ims Test Suite")
+}
+
+var _ = BeforeSuite(func() {
+	Expect(os.Getenv("TEST_HOST")).NotTo(BeZero(), "Please make sure TEST_HOST is set correctly.")
+})
+
+var _ = Describe("Metadata Server API testing", func() {
+	Context("When getting infrastructure Inventory API version", func() {
+		It("should return OK in the response and json response should match reference json", func() {
+			requestBody := []byte(`{"uriPrefix":"/o2ims-infrastructureInventory/v1","apiVersions":[{"version":"1.0.0"}]}`)
+			request, _ := http.NewRequest("GET", "https://"+os.Getenv("TEST_HOST")+"/o2ims-infrastructureInventory/api_versions", bytes.NewBuffer([]byte(requestBody)))
```

Comment:
The code to get the environment variable is repeated may times. You may want to get it in the `BeforeSuite` block, and save it to a global variable:

```go
var testHost string

var _ = BeforeSuite(func() {
        testHost = os.Getenv("TEST_HOST")
        ...
}
```

Then use it int he tests instead of repeating the call to `os.Getenv`: it reduces the chances that you use the wrong environment variable name.

Reply:
Done